easybuild/tools/filetools.py

```diff
@@ -763,6 +763,13 @@ def apply_patch(patch_file, dest, fn=None, copy=False, level=None):
     apatch = os.path.abspath(patch_file)
     adest = os.path.abspath(dest)
 
+    # Attempt extracting the patch if it does not end in .patch (may end in .gz, .zip, etc.)
+    if not apatch.endswith(".patch"):
+        workdir = tempfile.mkdtemp(prefix='eb-patch-')
+        _log.debug("Extracting the patch to: %s", workdir)
+        # extracting the patch
+        apatch = extract_file(apatch, workdir)
```

Comment:
@rjeschmi pointed out to me that `extract_file` actually doesn't return the path to the unpacked file (but to it's parent directory). This is probably also why one of the tests (i.e. `test_apply_patch `) now fails.

It probably *should* though, in case the unpacked file is again a file (rather than a directory, which is usually what we're dealing with)...